markdown/solace-k8s-operator/solace-k8s-operator.md

```diff
@@ -0,0 +1,214 @@
+
+```
+vi deploy/operator.yaml
+      containers:
+        - name: pubsubplus-operator
+          # Replace this with the built image name
+          image: kenbarr/pubsubplus-operator:v0.0.1 
+```
+
+Deploy the operator with specific service account and role binding.
+```
+kubectl create -f deploy/service_account.yaml
+kubectl create -f deploy/role.yaml
+kubectl create -f deploy/role_binding.yaml
+kubectl create -f deploy/operator.yaml
+```
+
+You can verify the operator container is deployed and running.
+``` 
+$ kubectl get deployment
+NAME                 DESIRED   CURRENT   UP-TO-DATE   AVAILABLE   AGE
+pubsubplus-operator       1         1         1            1           1m
+
+kubectl get pods
+NAME                                  READY   STATUS    RESTARTS   AGE
+pubsubplus-operator-5889cf85c-446kb   1/1     Running   0          3m36s
+```
+Finally run the operator to start the Solace cluster
+```
+kubectl apply -f deploy/crds/charts.helm.k8s.io_v1alpha1_pubsubplus_cr.yaml
+```
+
+Validate the Solace brokers are live and ready.
+```
+kubectl get pods
+NAME                                  READY   STATUS    RESTARTS   AGE
+example-pubsubplus-pubsubplus-0       1/1     Running   0          96s
+example-pubsubplus-pubsubplus-1       1/1     Running   0          96s
+example-pubsubplus-pubsubplus-2       1/1     Running   0          95s
+pubsubplus-operator-5889cf85c-446kb   1/1     Running   0          3m36s
+```
+Go on the primary broker with node ordinal 0 and validate "Redundancy Status" = "Up".  This means that all the brokers are syncronized and ready to recieve messages.
+```
+kubectl exec -it example-pubsubplus-pubsubplus-0 cli
+```
+```
+example-pubsubplus-pubsubplus-0> show redundancy
+
+Configuration Status     : Enabled
+Redundancy Status        : Up
+Operating Mode           : Message Routing Node
+Switchover Mechanism     : Hostlist
+Auto Revert              : No
+Redundancy Mode          : Active/Standby
+Active-Standby Role      : Primary
+Mate Router Name         : examplepubsubpluspubsubplus1
+ADB Link To Mate         : Up
+ADB Hello To Mate        : Up
+```
+Finally get the name of the loadbalancing Service that fronts the Solace cluster.  This will be the name used be applications connect to Solace. 
+```
+kubectl get services
+NAME                                      TYPE           CLUSTER-IP      EXTERNAL-IP   PORT(S)                                                                                                                                                                   AGE
+example-pubsubplus-pubsubplus             LoadBalancer   10.97.76.97     <pending>     2222:31348/TCP,8080:31717/TCP,1943:32170/TCP,55555:32597/TCP,55003:30106/TCP,55443:32318/TCP,8008:31054/TCP,1443:32713/TCP,5672:32610/TCP,1883:30046/TCP,9000:31317/TCP   5m4s
+```
+
+
+Using the test application found here: https://github.com/KenBarr/solace-node-sample to test, we see successful send and receipt of messages.
```

Comment:
`[https://github.com/KenBarr/solace-node-sample](https://github.com/KenBarr/solace-node-sample)` for hyperlink to be resolved